spec/States.html

```diff
@@ -923,9 +923,9 @@ <h3 id='choice-state'>Choice State</h3>
       <p>The values of the “And” and “Or” operators MUST be non-empty arrays
         of Choice Rules that MUST NOT contain “Next” fields; the
         “Next” field can only appear in a top-level Choice Rule.</p>
-      <p>The value of a “Not” operator MUST be a single Choice Rule, that
-        MUST NOT contain “Next” fields; the “Next” field can only
-        appear in a top-level Choice Rule.</p>
+      <p>You can create complex, nested Choice Rules using the "And", "Not",
```

Comment:
This is the current specification from the latest commit

```
A Boolean Expression is a JSON object which contains a field named "And", "Or", or "Not". If the field name is "And"
or "Or", the value MUST be an non-empty object array of Choice Rules, which MUST NOT contain "Next" fields; the 
interpreter processes the array elements in order, performing the boolean evaluations in the expected fashion, 
ceasing array processing when the boolean value has been unambiguously determined.

The value of a Boolean Expression containing a "Not" field MUST be a single Choice Rule, 
that MUST NOT contain a "Next" field; 
it returns the inverse of the boolean to which the Choice Rule evaluates.
```

Reply:
Maybe the second paragraph can be modified to:

```
The value of a Boolean Expression containing a "Not" field MUST be a single Choice Rule or a Boolean Expression, 
that MUST NOT contain a "Next" field; 
it returns the inverse of the boolean to which the Choice Rule evaluates.

Reply:
Adding "or a Boolean Expression" is redundant because "A Choice Rule is either a Boolean Express or a Data-test Expression" as defined in the spec in the section above. This was in the most recent update to the spec and I think clarifies the confusion.

What do you think? @mattsb42-aws